flink/src/main/java/org/apache/iceberg/flink/data/FlinkOrcWriter.java

```diff
@@ -0,0 +1,128 @@
+
+  private FlinkOrcWriter(RowType rowType, Schema iSchema) {
+    this.writer = (FlinkOrcWriters.StructWriter) FlinkOrcSchemaVisitor.visit(rowType, iSchema, new WriteBuilder());
+
+    List<LogicalType> fieldTypes = rowType.getChildren();
+    this.fieldGetters = Lists.newArrayListWithExpectedSize(fieldTypes.size());
+    for (int i = 0; i < fieldTypes.size(); i++) {
+      fieldGetters.add(RowData.createFieldGetter(fieldTypes.get(i), i));
+    }
+  }
+
+  public static OrcRowWriter<RowData> buildWriter(RowType rowType, Schema iSchema) {
+    return new FlinkOrcWriter(rowType, iSchema);
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public void write(RowData row, VectorizedRowBatch output) {
+    int rowId = output.size;
+    output.size += 1;
+
+    List<OrcValueWriter<?>> writers = writer.writers();
+    for (int c = 0; c < writers.size(); ++c) {
+      OrcValueWriter child = writers.get(c);
+      child.write(rowId, fieldGetters.get(c).getFieldOrNull(row), output.cols[c]);
+    }
+  }
+
+  private static class WriteBuilder extends FlinkOrcSchemaVisitor<OrcValueWriter<?>> {
+    private WriteBuilder() {
+    }
+
+    @Override
+    public OrcValueWriter<RowData> record(Types.StructType iStruct,
+                                          List<OrcValueWriter<?>> results,
+                                          List<LogicalType> fieldType) {
+      return FlinkOrcWriters.struct(results, fieldType);
+    }
+
+    @Override
+    public OrcValueWriter<?> map(Types.MapType iMap, OrcValueWriter<?> key, OrcValueWriter<?> value,
+                                 LogicalType keyType, LogicalType valueType) {
+      return FlinkOrcWriters.map(key, value, keyType, valueType);
+    }
+
+    @Override
+    public OrcValueWriter<?> list(Types.ListType iList, OrcValueWriter<?> element, LogicalType elementType) {
+      return FlinkOrcWriters.list(element, elementType);
+    }
+
+    @Override
+    public OrcValueWriter<?> primitive(Type.PrimitiveType iPrimitive, LogicalType flinkPrimitive) {
+      switch (iPrimitive.typeId()) {
+        case BOOLEAN:
+          return GenericOrcWriters.booleans();
+        case INTEGER:
+          return GenericOrcWriters.ints();
+        case LONG:
+          return GenericOrcWriters.longs();
+        case FLOAT:
+          return GenericOrcWriters.floats();
+        case DOUBLE:
+          return GenericOrcWriters.doubles();
+        case DATE:
+          return FlinkOrcWriters.dates();
+        case TIME:
+          return FlinkOrcWriters.times();
+        case TIMESTAMP:
+          Types.TimestampType timestampType = (Types.TimestampType) iPrimitive;
+          if (timestampType.shouldAdjustToUTC()) {
+            return FlinkOrcWriters.timestampTzs();
+          } else {
+            return FlinkOrcWriters.timestamps();
+          }
+        case STRING:
+          return FlinkOrcWriters.strings();
+        case UUID:
+          return GenericOrcWriters.uuids();
```

Comment:
for UUID type,  we flink should return `GenericOrcWriters.bytes()` instead of `GenericOrcWriters.uuids()`,  will fix this in next path.  The reader will also need to fix.